src/Cake.Common/Security/HashAlgorithmBuilder.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Security.Cryptography;
+
+namespace Cake.Common.Security
+{
+    /// <summary>
+    /// Creates a <see cref="System.Security.Cryptography.HashAlgorithm"/> instance by
+    /// <see cref="HashAlgorithm"/>.
+    /// </summary>
+    public sealed class HashAlgorithmBuilder : IHashAlgorithmBuilder
+    {
+        /// <summary>
+        /// Return a instance of <see cref="System.Security.Cryptography.HashAlgorithm"/> by
+        /// <see cref="HashAlgorithm"/> enum value.
+        /// </summary>
+        /// <param name="hashAlgorithm">Algorithm enum value.</param>
+        /// <returns>A <see cref="System.Security.Cryptography.HashAlgorithm"/> instance.</returns>
+        public System.Security.Cryptography.HashAlgorithm GetHashAlgorithm(HashAlgorithm hashAlgorithm)
```

Comment:
```suggestion
        public System.Security.Cryptography.HashAlgorithm CreateHashAlgorithm(HashAlgorithm hashAlgorithm)
```
If this goes public I'll suggest we call it `Create`.